src/Analyzers/CSharp/CodeFixes/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionCodeFixProvider.Rewriter.cs

```diff
@@ -102,11 +105,43 @@ private StatementSyntax GenerateVariableDeclaration(ExpressionSyntax switchExpre
             private SwitchExpressionArmSyntax GetSwitchExpressionArm(SwitchSectionSyntax node)
             {
                 return SwitchExpressionArm(
-                    pattern: GetPattern(SingleOrDefaultSwitchLabel(node.Labels), out var whenClauseOpt),
+                    pattern: GetPattern(node.Labels, out var whenClauseOpt),
                     whenClause: whenClauseOpt,
                     expression: RewriteStatements(node.Statements));
             }
 
+            private static PatternSyntax GetPattern(SyntaxList<SwitchLabelSyntax> switchLabels, out WhenClauseSyntax whenClauseOpt)
+            {
+                if (switchLabels.Count == 1)
+                    return GetPattern(switchLabels[0], out whenClauseOpt);
+
+                if (switchLabels.Any(label => IsDefaultSwitchLabel(label)))
+                {
+                    // original group had a catch-all label.  just convert to a discard _ to indicate the same.
+                    whenClauseOpt = null;
+                    return DiscardPattern();
+                }
+
+                // Multiple labels, and no catch-all merge them using an 'or' pattern.
+                var totalPattern = GetPattern(switchLabels[0], out var whenClauseUnused);
+                Debug.Assert(whenClauseUnused == null, "We should not have offered to convert multiple cases if any have a when clause");
+
+#if !CODE_STYLE
```

Comment:
not asking for a comment, I just don't quite get what's going on here - why is this only inside the code_style directive?

Reply:
because CODE_STYLE layer uses the last public release of Roslyn.  So it doesn't have access to these new compiler syntax apis.

Reply:
I am hoping to bump up CodeStyle layer to latest Roslyn bits this week (as decided in the last design meeting). I should be able to get rid of these conditional exclusions with that change.